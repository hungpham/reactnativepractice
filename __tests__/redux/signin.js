import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

//Testing: integrate with enzyme
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import {
	INITIAL_STATE,
	signInReducer,
	loginRequest,
	loginRequestSuccess,
	loginRequestFailed,
	REQUEST,
	SUCCESS,
	FAILED
} from 'redux/signin';

const MOCK_USERINFO = {
	trainee_id: 1,
	full_name: 'test',
	email: 'test@gmail.com',
	birthday: '1986-01-20',
	token: 'abcxyzwendsjkfjdsklfjkds'
};

const MOCK_ERROR = {message: 'testing failed'}

describe('SignIn Action Creators ', () => {
	it('creates a SUCCESS action', () => {
	  // You do it
	  expect(loginRequestSuccess(MOCK_USERINFO)).toEqual(
	    {
				type: SUCCESS,
				payload: MOCK_USERINFO
	    }
	  );
	  // Jest does it
	  expect(loginRequestSuccess(MOCK_USERINFO)).toMatchSnapshot();
	});

	it('creates a REQUEST action', () => {
	  // You do it
	  expect(loginRequest()).toEqual(
	    {
				type: REQUEST
	    }
	  );
	  // Jest does it
	  expect(loginRequest()).toMatchSnapshot();
	});

	it('creates a FAILED action', () => {
	  // You do it
	  expect(loginRequestFailed(MOCK_ERROR)).toEqual(
	    {
				type: FAILED,
				error: MOCK_ERROR.message
	    }
	  );
	  // Jest does it
	  expect(loginRequestFailed(MOCK_ERROR)).toMatchSnapshot();
	});
});

//TESTING Reducer
describe('SignIn Action Creators ', () => {
	it('should initiate loading', () => {
	  const stateAfter = {
			error: '',
			loading: true
	  };
	  expect(
	    signInReducer(INITIAL_STATE, {
	      type: REQUEST
	    })
	  ).toEqual(stateAfter);
	});

	it('should success signin', () => {
	  const stateAfter = {
			error: '',
			loading: false,
			user: MOCK_USERINFO
	  };
	  expect(
	    signInReducer(INITIAL_STATE, {
	      type: SUCCESS,
				payload: MOCK_USERINFO
	    })
	  ).toEqual(stateAfter);
	});

	it('should failed signin', () => {
	  const stateAfter = {
			error: MOCK_ERROR.message,
			loading: false
	  };
	  expect(
	    signInReducer(INITIAL_STATE, {
	      type: FAILED,
				error: MOCK_ERROR.message
	    })
	  ).toEqual(stateAfter);
	});

});
