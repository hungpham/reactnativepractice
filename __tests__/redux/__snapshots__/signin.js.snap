// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn Action Creators  creates a FAILED action 1`] = `
Object {
  "error": "testing failed",
  "type": "Failed",
}
`;

exports[`SignIn Action Creators  creates a REQUEST action 1`] = `
Object {
  "type": "Request",
}
`;

exports[`SignIn Action Creators  creates a SUCCESS action 1`] = `
Object {
  "payload": Object {
    "birthday": "1986-01-20",
    "email": "test@gmail.com",
    "full_name": "test",
    "token": "abcxyzwendsjkfjdsklfjkds",
    "trainee_id": 1,
  },
  "type": "Success",
}
`;
